sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -192,12 +245,23 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
         "partitions with value: " + dynamicPartitionKeys.keys.mkString("[", ",", "]"), ctx)
     }
 
-    InsertIntoTable(
-      UnresolvedRelation(tableIdent),
-      partitionKeys,
-      query,
-      ctx.OVERWRITE != null,
-      ctx.EXISTS != null)
+    (tableIdent, partitionKeys, ctx.OVERWRITE() != null, ctx.EXISTS() != null)
```

Comment:
Under `InsertOverwriteTableContext`, I think `ctx.OVERWRITE()` should not be null at all.

Reply:
You have `InsertOverwriteTableContext` and `InsertIntoTableContext` actually, that said you don't need an `overwrite` in `InsertTableParams`. You already know whether to overwrite before visiting them.

Reply:
updated.